### DISCUSSION TITLE
create_build task fails when retrieving the build_id inmediately

### DISCUSSION THREAD

**Discussion by dinigo** [apache/airflow#22033] [General]
### Apache Airflow Provider(s)

google

### Versions of Apache Airflow Providers

>>> pip show apache-airflow-providers-google
Name: apache-airflow-providers-google
Version: 6.3.0

### Apache Airflow version

2.2.3

### Operating System

Ubuntu 18.04.6 LTS

### Deployment

Composer

### What happened

When I trigger a build from the CloudBuildHook it always fails. I suspect due to the `build_id` not available when the Hook tries to retrieve it.

1. I call the `create_build()` hook method:
```python
    CloudBuildHook().create_build(
            build={
                # ...
            }
        )
```

2. [`create_build()` calls `_get_build_id_from_operation()`](https://github.com/apache/airflow/blob/main/airflow/providers/google/cloud/hooks/cloud_build.py#L165)
3. [`_get_build_id_from_operation()` raises "Could not retrieve Build Id from Operation"](https://github.com/apache/airflow/blob/main/airflow/providers/google/cloud/hooks/cloud_build.py#L78)

### What you expected to happen

For th

### How to reproduce

Call the `create_build()` hook method:
```python
    CloudBuildHook().create_build(
            build={
                # ...
            }
        )
```

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by dinigo** [comment]
I can see on the top of the file the variable `TIME_TO_SLEEP_IN_SECONDS = 5` unused. Could it be that this was used to "sleep" before retrieving the `_id`? 

https://github.com/apache/airflow/blob/main/airflow/providers/google/cloud/hooks/cloud_build.py#L33

**Comment by potiuk** [comment]
I suggest to take a look at the operator to see example of how hook is used. https://github.com/apache/airflow/blob/main/airflow/providers/google/cloud/operators/cloud_build.py

**Reply by danielpayoneer** [reply]
CloudBuildCreateBuildOperator failing for the same reason. but the build itself is executed.
